esd_services_api_client/boxer/_connector.py

```diff
@@ -32,11 +33,84 @@
     ExternalAuthBase,
     RefreshableExternalTokenAuth,
 )
-from esd_services_api_client.boxer._helpers import (
-    _iterate_user_claims_response,
-    _iterate_boxer_claims_response,
-)
-from esd_services_api_client.boxer._models import BoxerClaim, UserClaim, BoxerToken
+from esd_services_api_client.boxer._helpers import _iterate_user_claims_response
+from esd_services_api_client.boxer._models import BoxerToken, Claim
+
+
+class BoxerClaimConnector:
+    """
+    Boxer Claims API connector
+    """
+
+    def __init__(self, *, base_url: str, auth: Optional[BoxerTokenAuth] = None):
+        """Creates Boxer Claims connector, capable of managing claims
+        :param base_url: Base URL for Boxer Claims endpoint
+        :param auth: Boxer-based authentication
+        """
+        self.base_url = base_url
+        self.http = session_with_retries()
+        if auth and isinstance(auth, BoxerTokenAuth):
+            self.http.hooks["response"].append(auth.get_refresh_hook(self.http))
+        self.http.auth = auth
+
+    def get_claims(self, user_id: str, provider: str) -> Optional[Iterator[Claim]]:
+        """
+        Returns the claims assigned to the specified user_id and provider
+        """
+        response = self.http.get(f"{self.base_url}/claim/{provider}/{user_id}")
+        if response.status_code == 404:
+            return None
+        response.raise_for_status()
+        return _iterate_user_claims_response(response)
+
+    def add_user(self, user_id: str, provider: str) -> dict:
+        """
+        Adds a new user_id, provider pair
+        """
+        response = self.http.post(f"{self.base_url}/claim/{provider}/{user_id}")
+        response.raise_for_status()
+        return response.json()
+
+    def remove_user(self, user_id: str, provider: str) -> Response:
+        """
+        Removes the specified user_id, provider pair and assigned claims
+        """
+        response = self.http.delete(f"{self.base_url}/claim/{provider}/{user_id}")
+        response.raise_for_status()
+        return response
+
+    def add_claim(self, user_id: str, provider: str, claim: Claim) -> Optional[dict]:
+        """
+        Adds a new claim to an existing user_id, provider pair
+        """
+        claims = self.get_claims(user_id, provider)
+        if claims is not None:
+            payload = {"operation": "Insert", "claims": claim.to_dict()}
+            payload_json = json.dumps(payload)
```

Comment:
this deserves a class I think, inheriting from DataclassesJsonMixin

Reply:
I added ClaimPayload class